Mathlib/Data/List/Cycle.lean

```diff
@@ -273,106 +274,93 @@ theorem next_get : ∀ (l : List α) (_h : Nodup l) (i : Fin l.length),
         simp at this; simp [this] at hi'
       · rw [get_cons_succ]; exact get_mem _ _ _
 
-set_option linter.deprecated false in
-@[deprecated next_get (since := "2023-01-27")]
-theorem next_nthLe (l : List α) (h : Nodup l) (n : ℕ) (hn : n < l.length) :
-    next l (l.nthLe n hn) (nthLe_mem _ _ _) =
-      l.nthLe ((n + 1) % l.length) (Nat.mod_lt _ (n.zero_le.trans_lt hn)) :=
-  next_get l h ⟨n, hn⟩
-
-set_option linter.deprecated false in
-theorem prev_nthLe (l : List α) (h : Nodup l) (n : ℕ) (hn : n < l.length) :
-    prev l (l.nthLe n hn) (nthLe_mem _ _ _) =
-      l.nthLe ((n + (l.length - 1)) % l.length) (Nat.mod_lt _ (n.zero_le.trans_lt hn)) := by
-  cases' l with x l
-  · simp at hn
-  induction' l with y l hl generalizing n x
-  · simp
-  · rcases n with (_ | _ | n)
-    · simp [Nat.add_succ_sub_one, add_zero, List.prev_cons_cons_eq, List.length,
-        List.nthLe, Nat.succ_add_sub_one, zero_add, getLast_eq_get,
-        Nat.mod_eq_of_lt (Nat.succ_lt_succ l.length.lt_succ_self)]
-    · simp only [mem_cons, nodup_cons] at h
-      push_neg at h
-      simp only [List.prev_cons_cons_of_ne _ _ _ _ h.left.left.symm, List.length,
-        List.nthLe, add_comm, eq_self_iff_true, Nat.succ_add_sub_one, Nat.mod_self, zero_add,
-        List.get]
-    · rw [prev_ne_cons_cons]
-      · convert hl n.succ y h.of_cons (Nat.le_of_succ_le_succ hn) using 1
-        have : ∀ k hk, (y :: l).nthLe k hk = (x :: y :: l).nthLe (k + 1) (Nat.succ_lt_succ hk) := by
-          simp [List.nthLe]
-        rw [this]
-        congr
-        simp only [Nat.add_succ_sub_one, add_zero, length]
-        simp only [length, Nat.succ_lt_succ_iff] at hn
-        set k := l.length
-        rw [Nat.succ_add, ← Nat.add_succ, Nat.add_mod_right, Nat.succ_add, ← Nat.add_succ _ k,
-          Nat.add_mod_right, Nat.mod_eq_of_lt, Nat.mod_eq_of_lt]
-        · exact Nat.lt_succ_of_lt hn
-        · exact Nat.succ_lt_succ (Nat.lt_succ_of_lt hn)
-      · intro H
-        suffices n.succ.succ = 0 by simpa
-        rw [nodup_iff_nthLe_inj] at h
-        refine h _ _ hn Nat.succ_pos' ?_
-        simpa using H
-      · intro H
-        suffices n.succ.succ = 1 by simpa
-        rw [nodup_iff_nthLe_inj] at h
-        refine h _ _ hn (Nat.succ_lt_succ Nat.succ_pos') ?_
-        simpa using H
-
-set_option linter.deprecated false in
+theorem prev_get (l : List α) (_h : Nodup l) (i : Fin l.length) :
```

Comment:
I would prefer that we explicitly turn off the linter and comment about how it is failing at `h` than work around the issues.

Reply:
@Parcly-Taxel pushed these changes to my PR ... :-(

Reply:
I've deleted Jeremy's commit from this PR, so this comment now belongs on https://github.com/leanprover-community/mathlib4/pull/15965.

Reply:
(And I've implemented the suggestion there.)